daemon/containerd/image_exporter.go

```diff
@@ -20,18 +26,58 @@ import (
 //
 // TODO(thaJeztah): produce JSON stream progress response and image events; see https://github.com/moby/moby/issues/43910
 func (i *ImageService) ExportImage(ctx context.Context, names []string, outStream io.Writer) error {
+	platform := platforms.AllPlatformsWithPreference(cplatforms.Default())
 	opts := []archive.ExportOpt{
-		archive.WithPlatform(platforms.Ordered(platforms.DefaultSpec())),
 		archive.WithSkipNonDistributableBlobs(),
+
+		// This makes the exported archive also include `manifest.json`
+		// when the image is a manifest list. It is needed for backwards
+		// compatibility with Docker image format.
+		// The containerd will choose only one manifest for the `manifest.json`.
+		// Our preference is to have it point to the default platform.
+		// Example:
+		//  Daemon is running on linux/arm64
+		//  When we export linux/amd64 and linux/arm64, manifest.json will point to linux/arm64.
+		//  When we export linux/amd64 only, manifest.json will point to linux/amd64.
+		// Note: This is only applicable if importing this archive into non-containerd Docker.
+		// Importing the same archive into containerd, will not restrict the platforms.
+		archive.WithPlatform(platform),
```

Comment:
~~What's the behavior if the image we're saving only exists for _non_-default platforms? ie, a `linux/386`-only image on `linux/amd64` or similar?  We have a somewhat similar case with "shallow" pulls of a multi-arch image, where the image itself might support `linux/amd64` but we pulled some other platform specifically (and so `save` can only really save the one we have and it should thus be "default"), as noted by the `if` later. :sweat_smile:~~

Oh dur, that's what `AllPlatformsWithPreference` is for.  Sorry! :see_no_evil: